src/daft-connect/src/op/execute/root.rs

```diff
@@ -46,7 +46,7 @@ impl Session {
                 let cfg = Arc::new(DaftExecutionConfig::default());
                 let native_executor = NativeExecutor::from_logical_plan_builder(&optimized_plan)?;
 
-                let mut result_stream = native_executor.run(&pset, cfg, None)?.into_stream();
+                let mut result_stream = pin!(native_executor.run(&pset, cfg, None)?.into_stream());
```

Comment:
The kanal receiver doesn't provide a way to get an owned stream, so I manually made one using unfold, with a `recv.await` inside. Because of this await point I need to pin the stream.